### DISCUSSION TITLE
Issues trying to use Qt backend on macos

### DISCUSSION THREAD

**Discussion by jamesblacklock** [slint-ui/slint#565] [Q&A]
I'm getting the following error when I do a `cargo run` (seems to be just the `run`; `build` works fine):
```
dyld: Library not loaded: @rpath/QtCore.framework/Versions/A/QtCore
  Referenced from: /Users/<snip>/target/debug/budget
  Reason: image not found
```

Has anyone seen this error or have any familiarity with getting the sixtyfps qt backend working on macos?

**Comment by tronical** [comment]
The issue is that we can't easily set `rpath` on the binaries created with Rust. A workaround that you can use is to set the `DYLD_FRAMEWORK_PATH` environment variable to `/where/your/qt-install/lib`.

Nevertheless, there's maybe light at the end of the tunnel. While we also do run into https://github.com/rust-lang/cargo/issues/9562 for cdylibs we may be able to solve this by providing API for a build script for use on the application side and then use the new [`rustc-link-arg`](https://github.com/rust-lang/cargo/pull/9557/files) option in future releases.

I'll create a tracking issue for us to provide maybe API and docs for this :)

Edit: #566 created.

**Reply by jamesblacklock** [reply]
Environment variable workaround worked like a charm, thanks!
